### PR TITLE
Feature/logging event refactor

### PR DESCRIPTION
**Pull request by gsps1**
First step for https://issues.cask.co/browse/CDAP-7676.

We need to have LoggingEvent API in CDAP-watchdog-api, as we cannot depend on logback-core which has ILoggingEvent. 

Custom log processors will receive iterator of LoggingEvent.

Removed unused classes in cdap-watchdog-api.

Build:http://builds.cask.co/browse/CDAP-DUT5098

### CODE REVIEW

File: cdap-watchdog-api/src/main/java/co/cask/cdap/api/log/LoggerContextVO.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * the License.
+ */
+package co.cask.cdap.api.log;
+
+import java.util.Map;
+
+/**
+ * Replica of Logback LoggerContextVO
+ */
+public class LoggerContextVO {
```
Comment (poornachandra) [inline cdap-watchdog-api/src/main/java/co/cask/cdap/api/log/LoggerContextVO.java:23]: I don't think we use this anywhere today, let's not expose it in API until we know if this is really required.

File: cdap-watchdog-api/src/main/java/co/cask/cdap/api/log/LoggingEvent.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+package co.cask.cdap.api.log;
+
+import org.slf4j.Marker;
+import java.util.Map;
+
+/**
+ * LoggingEvent - replica for ILoggingEvent in LogBack.
+ */
+public interface LoggingEvent {
```
Comment (poornachandra) [inline cdap-watchdog-api/src/main/java/co/cask/cdap/api/log/LoggingEvent.java:25]: Would be good to call it `LogEvent`

File: cdap-watchdog-api/src/main/java/co/cask/cdap/api/log/LoggingEvent.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  /**
+   * Returns the MDC map. The returned value can be an empty map but not null.
+   */
+  Map<String, String> getMDCPropertyMap();
+
+  /**
+   * Synonym for [@link #getMDCPropertyMap}.
+   * @deprecated  Replaced by [@link #getMDCPropertyMap}
+   */
+  Map<String, String> getMdc();
```
Comment (poornachandra) [inline cdap-watchdog-api/src/main/java/co/cask/cdap/api/log/LoggingEvent.java:88]: Deprecated method can be removed

File: cdap-watchdog-api/src/main/java/co/cask/cdap/api/log/LoggingEvent.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+  long getTimeStamp();
+
+  void prepareForDeferredProcessing();
```
Comment (poornachandra) [inline cdap-watchdog-api/src/main/java/co/cask/cdap/api/log/LoggingEvent.java:92]: This method is not required, right? We can have only getters in this class.

File: cdap-watchdog-api/src/main/java/co/cask/cdap/api/log/StackTraceElementProxy.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.api.log;
+
+/**
+ * Replica of Logback StackTraceElementProxy
+ */
+public interface StackTraceElementProxy {
+  ClassPackagingData getClassPackagingData();
```
Comment (poornachandra) [inline cdap-watchdog-api/src/main/java/co/cask/cdap/api/log/StackTraceElementProxy.java:22]: Javadoc for all interface methods

**Review by poornachandra — COMMENTED** [review]
@gsps1 Few comments to address

**Comment by gsps1** [comment]
Closing this. on second thoughts, we realized its better to send ILoggingEvent than LoggingEvent. As user would most likely use logback api (Layout) for formatting the received logging event and that is possible only using ILoggingEvent object.

**Pull request closed without merging**